lib/ec2imgutils/ec2publishimg.py

```diff
@@ -156,6 +158,47 @@ def publish_images(self):
         images = self._get_images()
 
         for image in images:
+            if image['State'] != 'available':
+                waiter = self._connect().get_waiter('image_available')
+                loop_count = 0
+                while loop_count < self.wait_count:
+                    self.log.info('Current state of image %s is %s. \
+                                  Waiting up to 10 minutes for it to \
```

Comment:
This states we will wait up to 10 minutes, but there is no upper bounds check, thus this is actually not correct and with "--wait-count -> infinity" I could make the code wait forever. I am not advocating for an upper bounds check, just an accurate message.

Reply:
Robert,

According to the code, it does a 15 second wait (default) for 40 tries (default) before an error is created.

https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2.html#EC2.Waiter.ImageAvailable

Are we not using the same version of boto3 ?